### PR TITLE
Created CleanupTask for cleaning up md directories.

### PR DESCRIPTION
**Pull request by dotsdl**
Addresses #5.

This could be VERY dangerous if it isn't working as expected, so I have
not yet added this to any workflows generated by this library. For
workflows it is added to, it should probably be an opt-in kwarg to
include it.

It will delete 'files' in the fw_spec, which in this context
would be defined by BeaconTask. BeaconTask currently only puts the
'SCRATCHDIR' environment variable in this list of files, so that should
be the only thing deleted. It WOULD be catastrophic if somebody made
SCRATCHDIR a directory they didn't want to be deleted in full after a
run (like their home directory...).

### CODE REVIEW

**Comment by dotsdl** [comment]
Anybody in @Becksteinlab/owners or @Becksteinlab/mdworks want to take this one on for review? Interested in opinions to make it as stupid-proof and safe as possible. Automated deletion should always be done with great care.

**Comment by dotsdl** [comment]
My hyperparanoid opinion: we need to get continuous integration on Travis (requires #2) before I'd be fine with merging this. For now we deal with scratch directories that fill over time.

**Comment by orbeckst** [comment]
Yes, I agree, the autodelete is a bit too dangerous.

> On 19 May, 2016, at 16:34, David Dotson notifications@github.com wrote:
> 
> My hyperparanoid opinion: we need to get continuous integration on Travis (requires #2) before I'd be fine with merging this. For now we deal with scratch directories that fill over time.

## 

Oliver Beckstein \* orbeckst@gmx.net
skype: orbeckst  \* orbeckst@gmail.com

**Comment by orbeckst** [comment]
One way to make this safer is to drop a "semaphore" or token file into a directory and only delete if this file was found.

**Comment by dotsdl** [comment]
@orbeckst I like this idea. This looks like a way forward. This sounds sufficient, but any other ideas? Does that have any corner cases where it wouldn't be sufficiently safe?

**Comment by dotsdl** [comment]
Okay, now the `CleanupTask` is a whole lot safer. It only deletes specific files and directories given up by the `BeaconTask`, which is the same information used by `PullTask` to pull the resulting files back to the archive. For directories, `CleanupTask` deletes all the files inside, then removes the directory itself. It will **not** recursively delete the directory, so any directories inside will be safe from deletion.

Basically, it's really hard to mis-use `CleanupTask`, and even if you do, the damage is blunted by its very design. Already using this in my own work, so ready to merge.

**Comment by orbeckst** [comment]
Does it give any warning if the directory cannot be completely removed, e.g., because you created additional files that the `BeaconTask` did not know about? For instance, when Gromacs crashes it will often write out various `stepXXXX.pdb` files. 

Ideally, after a crash you wouldn't be deleting anything to facilitate the forensic analysis but I don't know if this is the case.

**Comment by dotsdl** [comment]
With the way the `BeaconTask` is used by `make_md_workflow`, it is given the directory in which the simulation is run on scratch on the remote resource. So, everything in that directory is pulled down, and that directory and all its file contents are deleted afterward by the separate `CleanupTask` Firework. We don't hardcode which files are pulled back.

**Comment by dotsdl** [comment]
This is safe. Been using it just fine now for nearly a week. I'm going to merge.
